webapp/channels/src/components/advanced_text_editor/use_editor_emoji_picker.tsx

```diff
@@ -57,62 +63,16 @@ const useEditorEmojiPicker = (
             return;
         }
 
-        let newMessage;
-        if (draft.message === '') {
-            newMessage = `:${emojiAlias}: `;
-            setCaretPosition(newMessage.length);
-        } else {
-            const {message} = draft;
-            const {firstPiece, lastPiece} = splitMessageBasedOnCaretPosition(caretPosition, message);
-
-            // check whether the first piece of the message is empty when cursor is placed at beginning of message and avoid adding an empty string at the beginning of the message
-            newMessage =
-                firstPiece === '' ? `:${emojiAlias}: ${lastPiece}` : `${firstPiece} :${emojiAlias}: ${lastPiece}`;
-
-            const newCaretPosition =
-                firstPiece === '' ? `:${emojiAlias}: `.length : `${firstPiece} :${emojiAlias}: `.length;
-            setCaretPosition(newCaretPosition);
-        }
-
-        handleDraftChange({
-            ...draft,
-            message: newMessage,
-        });
+        insertTextAtCaret(`:${emojiAlias}:`);
 
-        setEmojiSelected(true);
         setShowEmojiPicker(false);
-    }, [draft, caretPosition, handleDraftChange, setCaretPosition]);
+    }, [insertTextAtCaret]);
 
     const handleGifClick = useCallback((gif: string) => {
-        let newMessage: string;
-        if (draft.message === '') {
-            newMessage = gif;
-        } else if ((/\s+$/).test(draft.message)) {
-            // Check whether there is already a blank at the end of the current message
-            newMessage = `${draft.message}${gif} `;
-        } else {
-            newMessage = `${draft.message} ${gif} `;
-        }
-
-        handleDraftChange({
-            ...draft,
-            message: newMessage,
-        });
+        insertTextAtCaret(gif);
 
         setShowEmojiPicker(false);
-    }, [draft, handleDraftChange]);
-
-    // Focus textbox when the emoji picker closes
-    useDidUpdate(() => {
```

Comment:
This isn't needed any more because we now focus the textbox in `insertTextAtCaret` so that we can use `execCommand('insertText')`